decision_service/rlclientlib/bg_model_download.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+#include "restapi_data_transport.h"
+#include "data_callback_fn.h"
+namespace reinforcement_learning {
+  class error_callback_fn;
+}
+
+namespace reinforcement_learning { namespace model_management {
+  class bg_model_download {
+  public:
+    bg_model_download(i_data_transport* ptrans, data_callback_fn* pdata_cb);
+    ~bg_model_download();
+    bg_model_download(bg_model_download&& temp) noexcept;
+    bg_model_download& operator=(bg_model_download&& temp) noexcept;
+
+    int run_once(api_status* status) const;
+  private:
+    i_data_transport* _ptrans;
```

Comment:
>i_data_transport* _ptrans; [](start = 4, length = 26)

maybe use unique_ptr and safe and avoid the delete in the destructor?

Reply:
although destructor would be simpler (no delete), move constructor and assignment would be more complicated... no?

---
In reply to: [193485057](https://github.com/JohnLangford/vowpal_wabbit/pull/1495#discussion_r193485057) [](ancestors = 193485057)

Reply:
Moved out ownership of these pointers (in the next PR)

---
In reply to: [193517810](https://github.com/JohnLangford/vowpal_wabbit/pull/1495#discussion_r193517810) [](ancestors = 193517810,193485057)